src/api/Controllers/LayerController.cs

```diff
@@ -0,0 +1,99 @@
+﻿using BDMS.Models;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+
+namespace BDMS.Controllers;
+
+[ApiController]
+[Route("api/v{version:apiVersion}/[controller]")]
+public class LayerController : ControllerBase
+{
+    private readonly BdmsContext context;
+    private readonly ILogger logger;
+
+    public LayerController(BdmsContext context, ILogger<LayerController> logger)
+    {
+        this.context = context;
+        this.logger = logger;
+    }
+
+    /// <summary>
+    /// Asynchronously gets the <see cref="Layer"/>s, optionally filtered by <paramref name="profileId"/>.
+    /// </summary>
+    /// <param name="profileId">The id of the stratigraphy containing the layers to get.</param>
+    [HttpGet]
+    public async Task<ActionResult<IEnumerable<Layer>>> GetAsync(int? profileId = null)
+    {
+        var layers = context.Layers.AsQueryable();
+
+        if (profileId != null)
+        {
+            layers = layers.Where(l => l.StratigraphyId == profileId);
+
+            if (layers.Any())
+            {
+                return await layers.AsNoTracking().ToListAsync().ConfigureAwait(false);
+            }
```

Comment:
Falls diese Methode so stehen bleibt und nicht aufgeteilt wird:
PP: Diese `if`-Anweisung kann man wegoptimieren, in dem man nur den `NotFound();`-Fall hier behandelt. Am Ende der Methode werden die `layers` dann sowieso zurückgegeben.